client/asset/btc/electrum/network.go

```diff
@@ -0,0 +1,646 @@
+	Hex           string `json:"hex"`
+	Vin           []Vin  `json:"vin"`
+	Vout          []Vout `json:"vout"`
+	BlockHash     string `json:"blockhash,omitempty"`
+	Confirmations int32  `json:"confirmations,omitempty"` // probably uint32 ok because it seems to be omitted, but could be -1?
+	Time          int64  `json:"time,omitempty"`
+	BlockTime     int64  `json:"blocktime,omitempty"` // same as Time?
+	// Merkel // proto 1.5+
+}
+
+// TxID: f4b5fca9e2fa3abfe22ea82eece8e28cdcf3e03bb11c63c327b5c719bfa2df6f
+// vs
+// Hash: 81da4a53cbfd5c8b5a08709b8ccbbb28884287ed5a554ada387f7b08ae9d8f20 (?)
+
+// GetTransaction requests a transaction.
+func (sc *ServerConn) GetTransaction(ctx context.Context, txid string) (*GetTransactionResult, error) {
+	var resp GetTransactionResult
+	err := sc.Request(ctx, "blockchain.transaction.get", positional{txid, true}, &resp)
+	if err != nil {
+		return nil, err
+	}
+	return &resp, nil
+}
+
+// GetBlockHeaderResult represents the data returned by a request for a block
+// header(s) if the "cp_height" in the request was set and non-zero.
+// type GetBlockHeaderResult struct {
+// 	Branch []string `json:"branch"`
+// 	Header string   `json:"header"`
+// 	Root   string   `json:"root"`
+// }
+
+// BlockHeader requests the block header at the given height, returning
+// hexadecimal encoded serialized header.
+func (sc *ServerConn) BlockHeader(ctx context.Context, height uint32) (string, error) {
+	var resp string
+	err := sc.Request(ctx, "blockchain.block.header", positional{height}, &resp)
+	if err != nil {
+		return "", err
+	}
+	return resp, nil
+}
+
+// GetBlockHeadersResult represent the result of a batch request for block
+// headers via the BlockHeaders method. The serialized block headers are
+// concatenated in the HexConcat field, which contains Count headers.
+type GetBlockHeadersResult struct {
+	Count     uint32 `json:"count"`
+	HexConcat string `json:"hex"`
+	Max       uint32 `json:"max"`
+}
+
+// BlockHeaders requests a batch of block headers beginning at the given height.
+// The sever may respond with a different number of headers, so the caller
+// should check the Count field of the result.
+func (sc *ServerConn) BlockHeaders(ctx context.Context, startHeight, count uint32) (*GetBlockHeadersResult, error) {
+	var resp GetBlockHeadersResult
+	err := sc.Request(ctx, "blockchain.block.headers", positional{startHeight, count}, &resp)
+	if err != nil {
+		return nil, err
+	}
+	return &resp, nil
+}
+
+// SubscribeHeadersResult is the contents of a block header notification.
+type SubscribeHeadersResult struct {
+	Height int32  `json:"height"`
+	Hex    string `json:"hex"`
+}
+
+// SubscribeHeaders subscribes for block header notifications. There seems to be
+// no guarantee that we will be notified of all new blocks, such as when there
+// are blocks in rapid succession.
+func (sc *ServerConn) SubscribeHeaders(ctx context.Context) (*SubscribeHeadersResult, <-chan *SubscribeHeadersResult, error) {
```

Comment:
So I ended up not using the subscription machinery, but I'd really like to keep it in place in case it's needed.  Plus it rounds out ServerConn as an electrumx client.

Reply:
This pr pending changes? Will continue review when they are in.

Reply:
I have revisions based on your guys' review, but nothing for this method